File: Tutoriels/Java/POO/z0_Ressources/src/manager/GConfig.java
//===============================================
package manager;
//===============================================
import java.util.Map;
import java.util.HashMap;
import java.util.Iterator;
//===============================================
public class GConfig {
    //===============================================
    private Map<String, String> m_dataMap;
    //===============================================
    private GConfig() {
        m_dataMap = new HashMap<>();
    }
    //===============================================
    private static class GConfigHolder {       
        private final static GConfig m_instance = new GConfig();
    }
    //===============================================
    public static GConfig Instance() {
        return GConfigHolder.m_instance;
    }
    //===============================================
    public void setData(String key, String value) {
        m_dataMap.put(key, value);
    }
    //===============================================
    public String getData(String key) {
        return m_dataMap.get(key);
    }
    //===============================================
    public void showData() {
        Iterator lIterator = m_dataMap.entrySet().iterator();
        while(lIterator.hasNext()) {
            Map.Entry lData = (Map.Entry)lIterator.next();
            System.out.print(lData.getKey() + " = " + lData.getValue() + "\n");
        }
    }
    //===============================================
}
//===============================================

File: Tutoriels/Cpp/POO/z0_Ressources/src/manager/GConfig.cpp
//===============================================
#include "GConfig.h"
//===============================================
GConfig* GConfig::m_instance = 0;
//===============================================
GConfig::GConfig() {

}
//===============================================
GConfig::~GConfig() {

}
//===============================================
GConfig* GConfig::Instance() {
    if(m_instance == 0) {
        m_instance = new GConfig;
    }
    return m_instance;
}
//===============================================
void GConfig::setData(const string& key, const string& value) {
    m_dataMap[key] = value;
}
//===============================================
string GConfig::getData(const string& key) {
    return m_dataMap[key];
}
//===============================================
void GConfig::showData() {
    map<string, string>::iterator lIt;
    for(lIt = m_dataMap.begin(); lIt != m_dataMap.end(); lIt++) {
        cout << lIt->first << " = " << lIt->second << "\n";
    }
}
//===============================================
